### PR TITLE
Combine date and time picker demos

### LINKED ISSUE #6604: Standardize labelling among input widgets
Complex forms contain many types of input elements (checkboxes, dropdowns, pickers). It is difficult to use a consistent input labeling mechanism approved by Material. Take a look at this for instance:

image

The dropdown input labels "From" and "To" would match perfectly to what the input label will look like for "Location" once a value is entered. In reality, this would be a hack in the client app's code because Location's style and size is provided by the input widget while the "From"/"To" labels of the dropdowns are explicitly drawn by the app.

Leafy chose to fix this by disabling material labeling for all form elements and instead providing a consistent labeling mechanism. Unfortunately, this breaks when intrinsic padding is introduced. Examples:

- Top and bottom padding on Input
- Top and bottom padding on DropdownButton
- Start and end padding on Slider

It would be great if clients could specify the padding (if any) for these elements.

### PR DESCRIPTION
**Pull request by HansMuller**
The new demo also demonstrates how to create a form whose elements are all labeled and hinted like Input widgets.

- In the new example the first two rows are ordinary Input widgets.
- The next two rows are built with InputContainers that host what appear to be dropdowns but are in fact buttons. Tapping on the buttons pops up the date or time pickers.
- The last row is an InputContainer that hosts a Dropdown button.

Fixes #6604 

flutter_01

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/date_picker_demo.dart [hunk 1]
```diff
@@ -1,54 +0,0 @@
-// Copyright 2015 The Chromium Authors. All rights reserved.
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/date_picker_demo.dart]: looks like you missed time_picker_demo.dart

**Comment by Hixie** [comment]
What do you think of merging this and the other text input demo as well?

Also, what do you think of exposing the widgets you made here as actual widgets in the material layer?

LGTM.

File: examples/flutter_gallery/lib/demo/date_picker_demo.dart (hunk 1 above)
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/date_picker_demo.dart]: Yes I did, good catch!

**Comment by HansMuller** [comment]
I think you're right about providing InputDropdown and/or DateTimePicker as canned material widgets. Before doing that I want to resolve a few more of the outstanding dropdown issues.

**Pull request merged**